core/iwasm/common/wasm_c_api.c

```diff
@@ -2669,7 +2679,20 @@ wasm_func_call(const wasm_func_t *func, const wasm_val_vec_t *params,
     WASMExecEnv *exec_env = NULL;
     size_t param_count, result_count, alloc_count;
 
-    bh_assert(func && func->type && func->inst_comm_rt);
+    if (!func) {
+        return NULL;
+    }
+
+    if (!func->inst_comm_rt) {
+        wasm_name_t message = { 0 };
+        wasm_name_new_from_string(&message, "the function isn't linked");
```

Comment:
Do we need to check return value here?

Reply:
- if `wasm_name_new_from_string` failed, we have to return `NULL`
- a `NULL` message will lead `wasm_trap_new` returns `NULL` alone with a failure of creating a `wasm_trap_t`.

So I guess we can avoid checking here and let it go

Reply:
If wasm_name_new_from_string failed, it doesn't return NULL, but the message vector is empty:
wasm_name_new_from_string(&message, ...)
And here `trap = wasm_trap_new(func->store, &message);`  will create an empty trap, but not NULL:
```C
wasm_trap_t *
wasm_trap_new(wasm_store_t *store, const wasm_message_t *message)
{
    wasm_trap_t *trap;
    if (!store || !message) {
        return NULL;
    }
    if (!(trap = malloc_internal(sizeof(wasm_trap_t)))) {
        return NULL;
    }
    INIT_VEC(trap->message, wasm_byte_vec_new, message->size, message->data); 
    ==> message->size is 0, an empty trap is created.

    return trap;
...
```
Why not check the result of wasm_name_new_from_string like others, e.g.:
```C
        wasm_name_new_from_string(&module_name, module_name_rt);
        if (strlen(module_name_rt) && !module_name.data) {
            goto failed;
        }
```

Reply:
My bad explanation. I mean it doesn't matter if the `message` is empty or NULL.  The first priority is to raise a trap.  maybe I should change `wasm_trap_new`, like

`if (!store || !message)` => `if (!store)`

Reply:
OK, let trap be thrown firstly.